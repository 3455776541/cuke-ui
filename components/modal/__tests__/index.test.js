import React from "react";
import { render, shallow, mount } from "enzyme";
import assert from "power-assert";
import toJson from "enzyme-to-json";
import Modal from "../index";
import Button from "../../button";

describe("<Modal/>", () => {
  it("should render a <Modal/> components", () => {
    const wrapper = render(
      <div>
        <Modal title="基本使用" visible>
          <span>基本使用</span>
        </Modal>
        <Modal title="基本使用" okText="ok" cancelText="cancel" visible>
          <span>自定义文本 </span>
        </Modal>
        <Modal title="基本使用" visible footer={null}>
          <span>无 footer</span>
        </Modal>
        <Modal title="基本使用" visible closable centered>
          <span>居中</span>
        </Modal>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render custom footer width <Modal/> ", () => {
    const wrapper = render(
      <Modal
        title="基本使用"
        visible
        footer={[<a key="cancel">取消</a>, <a key="confirm">确定</a>]}
      >
        <span> 其实我是高仿 ant-design 的 </span>
      </Modal>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render default Button", () => {
    const wrapper = shallow(
      <Modal title="基本使用" visible>
        <span>基本使用</span>
      </Modal>
    );
    assert(wrapper.find(Button).length === 2);
  });

  it("should render custom ok button props", () => {
    const wrapper = render(
      <Modal visible okButtonProps={{ disabled: true }}>
        <span>基本使用</span>
      </Modal>
    );
    assert(wrapper.find(".cuke-button.btn-disabled").length === 1);
  });

  it("should render custom cancel button props", () => {
    const wrapper = render(
      <Modal visible cancelButtonProps={{ disabled: true }}>
        <span>基本使用</span>
      </Modal>
    );
    assert(wrapper.find(".cuke-button.btn-disabled").length === 1);
  });

  it("should render destroy reference", () => {
    const modal = Modal.confirm();
    assert(modal.destroy && modal.destroy instanceof Function);
  });

  it("should render confirm mode width <Modal/> ", () => {
    const wrapper = render(
      <Modal
        className="cuke-modal-confirm"
        showMask={false}
        closable={false}
        visible
      >
        <p>confirm</p>
      </Modal>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it.skip("should can not clicked with set maskClosable false", () => {
    const onCancelClick = jest.fn();
    const wrapper = mount(
      <Modal footer={null} visible={true} onCancel={onCancelClick}>
        <span>关闭回调</span>
      </Modal>
    );
    wrapper.find(".cuke-modal").simulate("click");
    assert(wrapper.props().visible === true);
    expect(onCancelClick).not.toHaveBeenCalled();
  });

  it("should can trigger onClose event", () => {
    const onClick = jest.fn();
    const onCancelClick = jest.fn();
    const wrapper = shallow(
      <Modal title="关闭回调" visible onOk={onClick} onCancel={onCancelClick}>
        <span>关闭回调</span>
      </Modal>
    );
    wrapper.find(".cuke-modal").simulate("click");
    expect(onClick).not.toHaveBeenCalled();
    expect(onCancelClick).not.toHaveBeenCalled();
  });
});
