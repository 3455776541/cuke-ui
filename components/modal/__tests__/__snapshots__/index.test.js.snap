// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> should render a <Modal/> components 1`] = `
<div>
  <div
    class="cuke-modal-mask cuke-modal-mask-show"
  />
  <div
    class="cuke-modal-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-modal cuke-modal-open"
      style="width:520px;z-index:999"
    >
      <section
        class="cuke-modal-header"
      >
        <h2
          class="cuke-modal-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-modal-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-modal-content"
      >
        <input
          class="cuke-input"
          placeholder="请输入"
          type="text"
        />
      </section>
      <section
        class="cuke-modal-footer"
      >
        <button
          class="cuke-button btn-default"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
        <button
          class="cuke-button btn-primary"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </section>
    </div>
  </div>
  <div
    class="cuke-modal-mask cuke-modal-mask-show"
  />
  <div
    class="cuke-modal-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-modal cuke-modal-open"
      style="width:520px;z-index:999"
    >
      <section
        class="cuke-modal-header"
      >
        <h2
          class="cuke-modal-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-modal-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-modal-content"
      >
        <input
          class="cuke-input"
          placeholder="请输入"
          type="text"
        />
      </section>
      <section
        class="cuke-modal-footer"
      >
        <button
          class="cuke-button btn-default"
          type="button"
        >
          <span>
            cancel
          </span>
        </button>
        <button
          class="cuke-button btn-primary"
          type="button"
        >
          <span>
            ok
          </span>
        </button>
      </section>
    </div>
  </div>
  <div
    class="cuke-modal-mask cuke-modal-mask-show"
  />
  <div
    class="cuke-modal-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-modal cuke-modal-open"
      style="width:520px;z-index:999"
    >
      <section
        class="cuke-modal-header"
      >
        <h2
          class="cuke-modal-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-modal-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-modal-content"
      >
        <input
          class="cuke-input"
          placeholder="请输入"
          type="text"
        />
      </section>
    </div>
  </div>
  <div
    class="cuke-modal-mask cuke-modal-mask-show"
  />
  <div
    class="cuke-modal-wrap cuke-modal-centered"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-modal cuke-modal-open"
      style="width:520px;z-index:999"
    >
      <section
        class="cuke-modal-header"
      >
        <h2
          class="cuke-modal-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-modal-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-modal-content"
      >
        <input
          class="cuke-input"
          placeholder="请输入"
          type="text"
        />
      </section>
      <section
        class="cuke-modal-footer"
      >
        <button
          class="cuke-button btn-default"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
        <button
          class="cuke-button btn-primary"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </section>
    </div>
  </div>
</div>
`;

exports[`<Modal/> should render confirm mode width <Modal/>  1`] = `
<div
  class="cuke-modal-wrap"
  role="dialog"
  tabindex="-1"
>
  <div
    class="cuke-modal cuke-modal-confirm cuke-modal-open no-title"
    style="width:520px;z-index:999"
  >
    <section
      class="cuke-modal-header"
    >
      <h2
        class="cuke-modal-title"
      />
    </section>
    <section
      class="cuke-modal-content"
    >
      <input
        class="cuke-input"
        placeholder="请输入"
        type="text"
      />
    </section>
    <section
      class="cuke-modal-footer"
    >
      <button
        class="cuke-button btn-default"
        type="button"
      >
        <span>
          取消
        </span>
      </button>
      <button
        class="cuke-button btn-primary"
        type="button"
      >
        <span>
          确定
        </span>
      </button>
    </section>
  </div>
</div>
`;

exports[`<Modal/> should render custom footer width <Modal/>  1`] = `
Array [
  <div
    class="cuke-modal-mask cuke-modal-mask-show"
  />,
  <div
    class="cuke-modal-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-modal cuke-modal-open"
      style="width:520px;z-index:999"
    >
      <section
        class="cuke-modal-header"
      >
        <h2
          class="cuke-modal-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-modal-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-modal-content"
      >
        <input
          class="cuke-input"
          placeholder="请输入"
          type="text"
        />
      </section>
      <section
        class="cuke-modal-footer"
      >
        <a>
          取消
        </a>
        <a>
          确定
        </a>
      </section>
    </div>
  </div>,
]
`;
